### DISCUSSION TITLE
Welcome to Legion-5600H-Hackintosh Discussions!

### DISCUSSION THREAD

**Discussion by kalkmann** [kalkmann/Legion-5600H-Hackintosh#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Parthkarmur** [comment]
Help me boot Ventura on Asus Vivobook 14 oled
Ryzen 5 5600h with Vega 7 Igpu no external gpu

**Reply by kalkmann** [reply]
You can try with https://github.com/NootInc/NootedRed - Instructions: https://nootinc.github.io/nred and https://nootinc.github.io/guide/gathering-files/kexts

You can receive help in forum https://forum.amd-osx.com/ and discord https://discord.gg/EfCYAJW

**Reply by ducviet5138** [reply]
@kalkmann Can you make a version for Legion which use iGPU based on your EFI? Thank you a lot.

**Reply by kalkmann** [reply]
I'm sorry, but I don't provide hackintoshing services. I take time out once a month and just catch up with what's new.

But there are several EFI's ready here on github to serve as a reference in your case (igpu oly). What is the brand and model of your laptop?

**Reply by ducviet5138** [reply]
Thank you for your reply. My laptop is also Legion-5600H but using 3050.

**Reply by kalkmann** [reply]
You can test with this EFI: https://github.com/ChgygLin/thinkbook16p-2021-5800H-EFI
Need to modify Kernel Patches from 8 to 6 cores: (first 4 items in **Kernel**-**Patch** in **config.plist**). At the beginning of the data "replace" field of each of them, replace BA0**8**... with BA0**6**...

**Comment by ChenZesen** [comment]
The built-in microphone in my computer can be recognized normally, but when I want to use it to record sound, it doesn't respond at all.
Later, I tried to adapt to AppleALC.kext, but it still didn't work. In addition, the built-in microphone in Windows 11 can use recording to confirm that there is no problem with the hardware.

**Reply by kalkmann** [reply]
Have you tested all possible layouts? You can try to build an applealc kext from the source, with the correct pin configurations. It is laborious and may have to be redone with each new version.

https://github.com/5T33Z0/AppleALC-Guides

https://brakintosh.gitbook.io/hackintosh/criando-layout-de-audio

https://github.com/F0x1c/AppleALC_Instructions

**Much easier, if the P2 microphone connection is essential in your workflow, is to buy a cheap usb audio dongle, it works independently of kexts:*

https://aliexpress.com/item/1005006010782653.html

**Comment by bdvela** [comment]
Hello, will you have any tutorial as a guide? I have the same Lenovo Legion 15ACH6 laptop but I don't know what the procedure to follow is.

**Reply by kalkmann** [reply]
Theoretically, the hardest part is done, assembling an EFI compatible with your laptop (the case with mine and others available).

Now you need to follow some guide to mount the installation media. You can follow https://dortania.github.io/OpenCore-Install-Guide/installer-guide/ and you will also find a lot of content on YouTube.

Read the credits here for this repository, there is a forum (and discord) with an active community that helps a lot.

I suggest installing on a free SSD for this purpose. Be patient! Good Luck!

**Reply by bdvela** [reply]
Hello, thanks for the suggestion. I was following the guide to create the bootable USB and when I restart the PC I access the OpenCore selector but it only appears to start Windows, I hit the space key and only "restart NVRAM" additionally appeared.
Do you know why the MacOS installer doesn't appear? any suggestions?

**Reply by kalkmann** [reply]
Where you make the usb, windows?

**Reply by bdvela** [reply]
Yes, this is my USB
![USB](https://github.com/kalkmann/Legion-5600H-Hackintosh/assets/55074395/2dd96eca-22a1-4c6c-8ee1-17f9d48e5883)

**Reply by kalkmann** [reply]
I've never mounted one on Windows. but search for an image of macos made by the olarila forum. I know you can download and write them using the Etcher tool, then you just replace the EFI folder with the one you are using.

**Comment by Gradi0us** [comment]
hello, I use a laptop with the same configuration as you, everything is the same but I can't use your EFI. I also have an efi version for myself but I need to turn off the xhc1 controller and it's quite annoying (I used it in the ventura version). If possible please check your telegram thank you very much.

**Reply by kalkmann** [reply]
Hello! Same BIOS version?

**Reply by Gradi0us** [reply]
yes, I have used the correct version you mentioned, this is the problem
![Screenshot 2024-04-01 065626](https://github.com/kalkmann/Legion-5600H-Hackintosh/assets/117055363/62e390df-93d5-4f8f-b273-2e9324265ad1)


)

**Reply by Gradi0us** [reply]
![image](https://github.com/kalkmann/Legion-5600H-Hackintosh/assets/117055363/6b2058f8-33d0-4cb1-8363-dce62808a649)
here is my bios version

**Reply by kalkmann** [reply]
From what I remember, I no longer had to disable XHC1 since Ventura (using the GUX-RyzenXHCIFix kext). Have you tried booting the installer on all back ports?

**Reply by Gradi0us** [reply]
I tried to install it with all ports (not disable xhc1) but it doesn't work, the problem same as the first image the macOS installer is hidden but when I try with  old EFI (since Ventura) the installer is displayed,idk what is going on @_@ 
Full spec:
legion 5 15ACH6A-5600h-6600M-intel ax210-WD SN750 500G + Cruicial P3 
Thanks

**Reply by kalkmann** [reply]
Try only with the GUX-RyzenXHCIFix kext and remove usbmap and ssdt-sleep

**Reply by Gradi0us** [reply]
that's not working

**Reply by kalkmann** [reply]
How you build the installation media?

**Comment by Gradi0us** [comment]
I downloaded the image from olarila and use Balenaetcher and replace efi
folder

Vào Th 2, 1 thg 4, 2024 lúc 18:06 kalkmann ***@***.***> đã
viết:

**Reply by kalkmann** [reply]
From what I understand from what you've written so far, you already had Ventura installed on your machine (using the EFI of the time that required disabling a USB controller). How you install the Ventura? Just redo what already worked once, then try using different EFIs to boot with the 2 USB controllers activated.

**Reply by Gradi0us** [reply]
I used to install Ventura last year but I must uninstall it because it's so inconvenient  USB and Bluetooth have been disabled but I see you fixed not turning the XHC 1 off and I tried with your efi to install it against but when boot from the USB I can't see the installer Sonoma while my old EFI from the Ventura can see it probably

**Reply by Gradi0us** [reply]
https://drive.google.com/file/d/1zfNWW9mqnJnYqeBLqD0pKh4JdG4mzav2/view?usp=drive_link
My old EFI can see the installer

**Reply by Gradi0us** [reply]
I downloaded your EFI and then used GenSMBIOS and copied it to the EFI partition on the USB, is that enough?

**Comment by kalkmann** [comment]
> I used to install Ventura last year but I must uninstall it because it's so inconvenient USB and Bluetooth have been disabled but I see you fixed not turning the XHC 1 off and I tried with your efi to install it against but when boot from the USB I can't see the installer Sonoma while my old EFI from the Ventura can see it probably

You don't need to uninstall MacOS to change the EFI and test new bios settings... With the system installed you can boot opencore from the pendrive, test new settings and when you are satisfied, replace the SSD's EFI.
Opencore (EFI) only manages the boot and performs patches, regardless of the system installed.

**Reply by Gradi0us** [reply]
But I uninstalled MacOs a long time ago to take advantage of that space @_@, above is the version I used, is there any way to fix it? I don't know why I can't boot from your EFI, do you think I should remap the USB?

**Reply by kalkmann** [reply]
I also have no idea why. If it's the same machine, in the same settings, I can only imagine it's the way the installation pendrive was created. I have never created via Windows. I always use gibmacos and createinstallmedia, all via Mac.

**Reply by Gradi0us** [reply]
I have tried creating an installer on Mac as well as resetting everything, but I encountered an issue when trying to install MacOS Sonoma. It gets stuck at the 12th minute and restarts. Do you have any ideas on how to fix this issue?
Thanks

**Reply by kalkmann** [reply]
Strange! When it's the same machine, it's assumed that everything goes equally well...
Which method did you use to create the pendrive?

**Reply by Gradi0us** [reply]
First, I downloaded the image from the forum but I heard you created the installer from MacOs so I also downloaded it directly from Apple but it still doesn't work. I don't use your EFI, I use kext and acpi similar to yours. , I tried with your efi but cannot boot like the first picture I described

**Reply by kalkmann** [reply]
If everything was done the same and the hardware was the same, there would be no reason to have a problem.
If you assume that you have experience in hackintoshing and it cannot be a problem with the method/software, then you could only assume a failure in some hardware (pendrive, etc.).

**Reply by Gradi0us** [reply]
inally, I managed to install it successfully, but currently, the Speaker, microphone, as well as FaceTime and iMessage, still don't work. Do you have any way to fix it? (I have added the AppleALC kext as well as adjusted the layout-id to 21, but it seems it's not working)

**Reply by kalkmann** [reply]
You didn't comment the solution you found to install :|

About iServices: https://dortania.github.io/OpenCore-Post-Install/universal/iservices.html

Confirm your laptop's audio codec (Hackintool's sound tab). Upload your config.plist here.

**Reply by Gradi0us** [reply]
Well.. about how I fixed the error, I'm not even sure anymore. I updated the BIOS to version 37 and reset the BIOS. I disabled the Igpu, and miraculously, I could boot normally. Here's the config file I'm using, please take a look and help me out. I think both you and I have the same Realtek ALC 287. I've tried layouts 21 and 13, but it's still the same. I hope you can point out where I'm wrong. 
And I also have another issue, although it doesn't affect much.When I boot It only displays text, and I would like it to show icons as well. I hope you understand what I mean, as I'm not sure how to describe it anymore.
Thank you very much
[config.plist.zip](https://github.com/kalkmann/Legion-5600H-Hackintosh/files/14897795/config.plist.zip)

**Reply by kalkmann** [reply]
What do you mean by "onboard cpu card"?

About the audio layout, you are declaring it 2 times with different values (choose one or other). In "DeviceProperties" it is correct. In the boot-args, I don't think so.

<img width="906" alt="image" src="https://github.com/kalkmann/Legion-5600H-Hackintosh/assets/8891448/0bc8ad69-1710-4497-b54f-117cf934d570">

**Reply by Gradi0us** [reply]
onboard cpu card == igpu 😂

I'will replace boot-args,Can I use your boot arg into my config?

**Reply by kalkmann** [reply]
<img width="293" alt="image" src="https://github.com/kalkmann/Legion-5600H-Hackintosh/assets/8891448/e7ca60ad-f8d2-4fe9-8879-70c5af7c2aa8">

🤷‍♂️

For audio work, a reference in DeviceProperties layout would be enough.

The arguments I use are specific to the EFI I use, they are optional, used in conjunction with kexts to fix and increase compatibility of native functions. The system can boot without any arguments, smoothly.

**Reply by Gradi0us** [reply]
I forgot that when updating the BIOS, all settings revert to default mode(The IGPU is enable).The EFI I'm using is similar to 99.99999999% except for the config.plist.

**Reply by Gradi0us** [reply]
I replace boot-agr and the speaker is working but the microphone built in still dead...

**Reply by Gradi0us** [reply]
I replaced the layout ID with 13 and everything works fine. Thank you for your help.
😍😍
